hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -817,7 +817,7 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	// Reconcile the CAPI Cluster resource
 	// In the None platform case, there is no CAPI provider/resources so infraCR is nil
 	if infraCR != nil {
-		capiCluster := controlplaneoperator.CAPICluster(controlPlaneNamespace.Name, hcluster.Spec.InfraID)
+		capiCluster := controlplaneoperator.CAPICluster(controlPlaneNamespace.Name, hcluster.Name)
```

Comment:
CAPI cluster name is intentionally infraID since this is used by the AWS providers for tagging VMs https://github.com/openshift/hypershift/pull/95

Reply:
ok but this change makes us not have an infraID when the provider is not AWS. Needing to name it like that in the AWS case makes this change significantly more complicated. I suppose we can add a `CapiClusterName` method to the `Platform` interface that returns the infraID in case of AWS and the hcluster name otherwise?

Reply:
That'd be an option, we'd need to reference it here as well 
That seems a good idea https://github.com/openshift/hypershift/pull/839/files#diff-b5cde720842b19b53da3f454d827e27327c4559b2e4849906dac728586f51eb3L1757

Reply:
At that point I honestly question the value of this change, that is a lot of complexity leaking into all providers (since they have to implement that new method) just to be able to nest a field differently. Before we do that I'd just rather leave it as is and leave a todo to maybe rethink this when doing a v2 of our api after we have more providers.

Reply:
I agree suggestion above would become convoluted but I'm fairly sure we can find a cleaner implementation if we decide to move the field. To make that decision let's keep the focus on the desired UX:
>Before we do that I'd just rather leave it as is and leave a todo to maybe rethink this when doing a v2 of our api after we have more providers.

That make sense but I see it kind of the other way around: unless there's a real reason for the field to be generic (trying to clarify that with my question [below](https://github.com/openshift/hypershift/pull/839#issuecomment-1011535161)) it should be aws specific (and the abstraction for now should be a programatic an impl detail for as to best accommodate our code) and only when more providers express a need we should consider abstract it away and make a generic API user input.

I'll bring this up once again next meeting to do some last team brainstorming and make a decision.